### DISCUSSION TITLE
compilation issue

### DISCUSSION THREAD

**Discussion by vdeepankar2710** [cpeditor/cpeditor#1035] [Q&A] [answered]
[13:49:45] [Compiler] [Failed to start compilation: Failed to start the compiler. Please check Preferences->Language->C++->C++ Commands->C++ Compile Command or add the compiler in the PATH environment variable.]

I receive the above message whenever I try to compile my code in c++. 
My g++ version is 11.2.0, c++ compile command is set to "c++ -Wall" in the preferences and executable file path is set to "${tmpdir}/${basename}", still I get this error. Someone please help.

**Comment by ouuan** [comment, marked as answer]
Is your gcc in the PATH environment variable?

**Reply by vdeepankar2710** [reply]
yes it was added, I have figured out that all the basic components of mingw was not installed. Now it works and compiles fine. Thanks.

**Comment by Akatkit** [comment]
Can someone sends me a solution to this I don't know how to fix it?

**Reply by ouuan** [reply]
No one can help you with no information provided. I have to ask you a bunch of questions as a result.

Have you installed gcc? How did you install it? Did you set the PATH environment variable for gcc? What's your OS? Can you run gcc in the terminal?
